### PR TITLE
[Doc]: Add document of TensorRT and onnxruntime custom ops

### PR DESCRIPTION
**Pull request by grimoire**
This PR adds detailed documents of TensorRT and onnxruntime custom ops.

The documents share the same format as [onnx document](https://github.com/onnx/onnx/blob/master/docs/Operators.md#aionnx-default-1).
Headers like `Description` `Parameters` use HTML tags instead of markdown to avoid [MD024](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md024---multiple-headers-with-the-same-content) check.

Note that sphinx on my device can not convert relative path like `./tensorrt_custom_ops.md#mmcvroialign` in `tensorrt_plugin.md` to HTML. I have no idea how to fix it.

### CODE REVIEW

File: docs/onnxruntime_custom_ops.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+<dd>Output boxes and scores. 2-D tensor of shape (num_valid_boxes, 5), [[x1, y1, x2, y2, score], ...]. num_valid_boxes is the number of valid boxes.</dd>
+<dt><tt>indices</tt>: T</dt>
+<dd>Output indices. 1-D tensor of shape (num_valid_boxes, ).</dd>
+</dl>
+
+<h3>Type Constraints</h3>
+
+- T:tensor(float32)
+
+## RoiAlign
```
Comment (RunningLeon) [inline docs/onnxruntime_custom_ops.md:50]: RoIAlign

File: docs/onnxruntime_op.md [hunk 2]
```diff
@@ -15,10 +15,11 @@
 
 ## List of operators for ONNX Runtime supported in MMCV
 
-| Operator |  CPU  |  GPU  |                                                Note                                                 |
-| :------: | :---: | :---: | :-------------------------------------------------------------------------------------------------: |
-| SoftNMS  |   Y   |   N   | commit [94810f](https://github.com/open-mmlab/mmcv/commit/94810f2297871d0ea3ca650dcb2e842f5374d998) |
-| RoiAlign |   Y   |   N   |                                                None                                                 |
+|                    Operator                    |  CPU  |  GPU  |                                                Note                                                 |
+| :--------------------------------------------: | :---: | :---: | :-------------------------------------------------------------------------------------------------: |
+|  [SoftNMS](onnxruntime_custom_ops.md#softnms)  |   Y   |   N   | commit [94810f](https://github.com/open-mmlab/mmcv/commit/94810f2297871d0ea3ca650dcb2e842f5374d998) |
```
Comment (RunningLeon) [inline docs/onnxruntime_op.md:20]: we'd better remove commit info from this table and update with the version of mmcv once later.

File: docs/onnxruntime_custom_ops.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+<h3>Type Constraints</h3>
+
+- T:tensor(float32)
+
+## RoIAlign
+
+<h3>Descriptions</h3>
+
+Perform ROIAlign on output feature, used in bbox_head of most two-stage detectors.
```
Comment (RunningLeon) [inline docs/onnxruntime_custom_ops.md:54]: RoIAlign

File: docs/onnxruntime_custom_ops.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+| `float` | `sigma`         | hyperparameter for gaussian method                             |
+| `float` | `min_score`     | score filter threshold                                         |
+| `int`   | `method`        | method to do the nms, (0: `naive`, 1: `linear`, 2: `gaussian`) |
+| `int`   | `offset`        | `boxes` width or height is (x2 - x1 + offset). (0 or 1)        |
+
+<h3>Inputs</h3>
+
+<dl>
+<dt><tt>boxes</tt>: T</dt>
+<dd>Input boxes. 2-D tensor of shape (N, 4). N is the batch size.</dd>
```
Comment (RunningLeon) [inline docs/onnxruntime_custom_ops.md:32]: Shouldn't `N` be the number of boxes?

File: docs/onnxruntime_custom_ops.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+Perform ROIAlign on output feature, used in bbox_head of most two-stage detectors.
+
+<h3>Parameters</h3>
+
+| Type    | Parameter        | Description                                                                                            |
+| ------- | ---------------- | ------------------------------------------------------------------------------------------------------ |
+| `int`   | `output_height`  | height of output roi                                                                                   |
+| `int`   | `output_width`   | width of output roi                                                                                    |
+| `float` | `spatial_scale`  | scale the input boxes by this                                                                          |
```
Comment (RunningLeon) [inline docs/onnxruntime_custom_ops.md:62]: `used to scale the input boxes`

File: docs/onnxruntime_custom_ops.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+| `int`   | `sampling_ratio` | number of inputs samples to take for each output sample. 0 to take samples densely for current models. |
```
Comment (RunningLeon) [inline docs/onnxruntime_custom_ops.md:63]: inputs -> input
`0` means to take samples densely for current models.

File: docs/onnxruntime_custom_ops.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+<h3>Type Constraints</h3>
+
+- T:tensor(float32)
+
+## NMS
+
+<h3>Descriptions</h3>
+
+Filter out boxes has high IOU overlap with previously selected boxes.
```
Comment (RunningLeon) [inline docs/onnxruntime_custom_ops.md:91]: IoU

File: docs/onnxruntime_custom_ops.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+<h3>Parameters</h3>
+
+| Type    | Parameter       | Description                                                                                                      |
+| ------- | --------------- | ---------------------------------------------------------------------------------------------------------------- |
+| `float` | `iou_threshold` | The threshold for deciding whether boxes overlap too much with respect to IOU. Value range [0, 1]. Default to 0. |
```
Comment (RunningLeon) [inline docs/onnxruntime_custom_ops.md:97]: IoU

File: docs/tensorrt_custom_ops.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+Perform ROIAlign on output feature, used in bbox_head of most two stage
+detectors.
+
+<h3>Parameters</h3>
+
+| Type    | Parameter        | Description                                                                                            |
+| ------- | ---------------- | ------------------------------------------------------------------------------------------------------ |
+| `int`   | `output_height`  | height of output roi                                                                                   |
+| `int`   | `output_width`   | width of output roi                                                                                    |
+| `float` | `spatial_scale`  | scale the input boxes by this                                                                          |
```
Comment (RunningLeon) [inline docs/tensorrt_custom_ops.md:27]: used to scale the input boxes

File: docs/onnxruntime_op.md [hunk 10]
```diff
@@ -15,10 +15,11 @@
...
 ## List of operators for ONNX Runtime supported in MMCV
 
-| Operator |  CPU  |  GPU  |                                                Note                                                 |
-| :------: | :---: | :---: | :-------------------------------------------------------------------------------------------------: |
-| SoftNMS  |   Y   |   N   | commit [94810f](https://github.com/open-mmlab/mmcv/commit/94810f2297871d0ea3ca650dcb2e842f5374d998) |
-| RoiAlign |   Y   |   N   |                                                None                                                 |
+|                    Operator                    |  CPU  |  GPU  | MMCV Releases |
+| :--------------------------------------------: | :---: | :---: | :-----------: |
+|  [SoftNMS](onnxruntime_custom_ops.md#softnms)  |   Y   |   N   |     1.2.3     |
+| [RoiAlign](onnxruntime_custom_ops.md#roialign) |   Y   |   N   |     1.2.5     |
```
Comment (RunningLeon) [inline docs/onnxruntime_op.md:21]: RoIAlign

File: docs/onnxruntime_op.md [hunk 11]
```diff
@@ -15,10 +15,11 @@
...
+|      [NMS](onnxruntime_custom_ops.md#nms)      |   Y   |   N   |     1.2.7     |
```
Comment (RunningLeon) [inline docs/onnxruntime_op.md:22]: Could add doc for `grid_sampler` with ONNX Runtime as well since #916 has been merged.

**Review by ZwwWayne — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by hellock** [comment]
We can set the argument `allow_different_nesting` to true to avoid MD024, as described in the [doc](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md024---multiple-headers-with-the-same-content).

**Pull request merged**